
## Output Files

The output files in this folder are intermediate files towards the final CODEML analysis. In order to run CODEML, a multiple sequence alignment and phylogenetic tree is needed for each gene.

### Multiple sequence alignments

Sequences are aligned using pre-set parameters of [Clustal Omega 1.2.2](https://www.ebi.ac.uk/Tools/msa/clustalo/), and output is used as input for [Gblocks 0.91b](http://molevol.cmima.csic.es/castresana/Gblocks_server.html) for a gap-free alignment. 

Parameters for Gblocks: sequence type (-t) is set to codons, and no minimum or maximum number of contiguous conserved/nonconserved positions are set. Gap positions are not allowed. In order to do downstream phylogenetic and selection analysis, the Gblocks sequence alignment is kept in two file formats: the original FASTA, and a nexus file generated by converting the FASTA.

### Phylogenetic trees

Trees are constructed in MrBayes version pre-3.2.7 using the Gblocks alignment as input. The follow commands are used in the command-line interface to set the substitution model (Nst), and gamma distribution model with invariant positions: 

```sh

$ execute <filename>
#If file is properly execute, the prompt will indicate that the file was read, at which point parameters can be set
$ lset Nst=2 
$ lset Rate=Invgamma
$ mcmc
#The program is automatically set to run 1,000,000 generations and 4 MCMC chains; once tree is complete, discard a percentage of total treees from beginning of chain
$ sumt relburnin=yes burninfrac=0.25

```




 


